libtcmu_log.c

```diff
@@ -245,29 +239,18 @@ log_internal(int pri, struct tcmu_device *dev, const char *funcname,
 	vsnprintf(buf + n, LOG_MSG_LEN - n, fmt, args);
 
 	/*
-	 * Bypass the ringbuffer for some cases,
-	 * such as stdout and log file
+	 * Avoid overflowing the log buf with SCSI CDBs.
 	 */
-	log_output(pri, buf, true);
-
-	if (pri >= TCMU_LOG_DEBUG_SCSI_CMD)
-		goto unlock;
+	if (pri < TCMU_LOG_DEBUG_SCSI_CMD)
+		log_queue_msg(tcmu_logbuf, pri, buf);
```

Comment:
BTW, before we will only output the TCMU_LOG_DEBUG_SCSI_CMD logs to stderr, and after your changes, it will output to the tcmu-runner.log file. Is this what you are trying to do, right ?

Before why I just output this to stderr was trying to avoid the tcmu-runner.log file increase to large. But I am okay with your change here or maybe we need a log file rotate ?

Reply:
Yeah, that sounds nice. :-)